treeherder/webapp/api/performance_data.py

```diff
@@ -468,7 +475,7 @@ def list(self, request):
 
         signature_ids = [item['id'] for item in list(self.queryset)]
 
-        data = (PerformanceDatum.objects.select_related('push', 'repository')
+        data = (PerformanceDatum.objects.select_related('push', 'repository', 'job')
```

Comment:
We don't need to retrieve the entire job object with `select_related` since we're accessing `id` via the foreign key value `job_id` and that's all we need (here and on line 455).

Reply:
That would require an extra filter by revision on line 464 at `if signature:`. We require the ids of all jobs triggered against that revision. I'm not yet sure how a query like that is done.

Reply:
> That would require an extra filter by revision on line 464 at `if signature:`. We need the ids of all jobs triggered against that revision.

Your use of `job_id` is correct, what I'm saying is you do not need to add `job` to `select_related()` since we are only grabbing the id's, of which there is already an index on there:
https://docs.djangoproject.com/en/2.1/topics/db/optimization/#use-foreign-key-values-directly

We don't need to fetch the entire Job object (because it contains data we don't need).
https://docs.djangoproject.com/en/2.1/ref/models/querysets/#django.db.models.query.QuerySet.select_related

> @sarah-clements I'm not yet sure how a query like that is done, but will give it a try. Reason: `select_related()` does join `performance_signature` with `push` table, but how can I filter by `push`'s revision?

I don't follow what you're getting at. Can you please try explaining it differently?

Reply:
> > That would require an extra filter by revision on line 464 at `if signature:`. We need the ids of all jobs triggered against that revision.
> 
> Your use of `job_id` is correct, what I'm saying is you do not need to add `job` to `select_related()` since we are only grabbing the id's, of which there is already an index on there:
> https://docs.djangoproject.com/en/2.1/topics/db/optimization/#use-foreign-key-values-directly

I see now. Thanks for explaining this!